cmd/devstream/list.go

```diff
@@ -21,7 +23,26 @@ var listPluginsCMD = &cobra.Command{
 	Long: `This command lists all of the plugins.
 Examples:
   dtm list plugins`,
-	Run: listPluginsCMDFunc,
+	ValidArgsFunction: compListPlugins,
+	Run:               listPluginsCMDFunc,
+}
+
+func compListPlugins(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) == 0 {
+		return list.PluginsNameSlice(), cobra.ShellCompDirectiveNoFileComp
+	}
+
+	if len(args) == 1 {
+		var choices []string
+		for _, pluginName := range list.PluginsNameSlice() {
+			if strings.HasPrefix(pluginName, args[0]) {
```

Comment:
The purpose of `dtm list plugins` is to show all available plugins, so why filter the specified plugins here?

Reply:
same as `dtm list plugins -r`, filter plugin by regex.

Reply:
Got it!

Reply:
Yes, do [TAB][TAB] to match or complete. complete is always like this.

Reply:
> Yes, do [TAB][TAB] to match or complete. complete is always like this.

Yeah, but plugins here are not args or subcommands. It's just like `kubectl plugin list`.

Reply:
I don't think this is the problem. When the user wants to know how many plugins are related to `argocd`, they can simply do `dtm list plugins argocd[TAB][TAB]` to filter them out. I believe this is also the purpose of `dtm list plugins -r`.

Reply:
@prodanlabs the only ways to use the `list plugins` subcommand are:
- `dtm list plugins` which shows all the plugins
- `dtm list plugins -r argo` which shows the plugins whose names match "argo"

So, even if the user types `dtm list plugins argo [tab][tab]` and it shows `dtm list plugins argocd`, it's not useful, because the user can't execute this command.

I think that's why @imxw thought [this](https://github.com/devstream-io/devstream/pull/650/files#diff-3857dee2ec9cf176ffd4d69f926a87374ed4c37ce8a5db3ce5ac6ce8b737daa0R25) isn't necessary.